torch/_utils.py

```diff
@@ -104,6 +107,8 @@ def _accumulate(iterable, fn=lambda x, y: x + y):
 def _flatten_tensors(tensors):
     """Flatten tensors into a single contiguous 1D buffer"""
     if len(tensors) == 1:
```

Comment:
I didn't see this code get adjusted for `len(tensors) != 1` and you got sparse tensors. Does that need fixing?

Reply:
The way the code works now is that when sparse tensors are given to `_flatten_tensor`, `tensors` will only be a single sparse tensor. I will update the comment on this!

Reply:
This is so because `reduce_add` adds each chunk from `_take_tensor` together, which is not applicable to sparse tensors. So I didn't to the case when `len(tensors) != 1`. But upon thinking about it, I do think that it should be supported. It will be at least useful for `broadcast_coalesced`.

Reply:
To me this looks incorrect. Is it really needed anywhere or can we remove it?